homeassistant/components/remote/__init__.py

```diff
@@ -110,3 +125,15 @@ def async_send_command(self, command, **kwargs):
         """
         return self.hass.async_add_job(ft.partial(
             self.send_command, command, **kwargs))
+
+    def learn_command(self, **kwargs):
+        """Learn a command from a device."""
+        raise NotImplementedError()
+
+    def async_learn_command(self, **kwargs):
+        """Learn a command from a device.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(ft.partial(
```

Comment:
Use `await self.hass.async_add_executor_job`, instead of returning the task.

Reply:
Like this?

```py
def async_learn_command(self, **kwargs):
    """Learn a command from a device.

    This method must be run in the event loop.
    """
    await self.hass.async_add_executor_job(ft.partial(
        self.learn_command, **kwargs))
```

Okay, I'll update it, but what's the idea behind it? I got this snippet from async_send_command, should I change it as well?

Reply:
We can change async_send_command separately. We define coroutines for the async method interface.

Reply:
Done.